x-pack/test/functional/apps/search_playground/playground_overview.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OpenAISimulator } from '@kbn/actions-simulators-plugin/server/openai_simulation';
+import { isServerlessKibanaFlavor } from '@kbn/security-solution-plugin/scripts/endpoint/common/stack_services';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+const indexName = 'basic_index';
+const esArchiveIndex = 'test/api_integration/fixtures/es_archiver/index_patterns/basic_index';
+
+async function* createOpenAIConnector(configService, supertest) {
+  const simulator = new OpenAISimulator({
+    returnError: false,
+    proxy: {
+      config: configService.get('kbnTestServer.serverArgs'),
+    },
+  });
+  yield simulator;
+
+  const config = {
+    apiProvider: 'OpenAI',
+    defaultModel: 'gpt-4',
+    apiUrl: await simulator.start(),
+  };
+  // eslint-disable-next-line prefer-const
+  let connector;
+
+  yield async () => {
+    if (connector) {
+      await supertest
+        .delete(`/api/actions/connector/${connector.id}`)
+        .set('kbn-xsrf', 'foo')
+        .set('x-elastic-internal-origin', 'foo')
+        .expect(204);
+    }
+
+    await simulator.close();
+  };
+
+  connector = (
+    await supertest
+      .post('/api/actions/connector')
+      .set('kbn-xsrf', 'foo')
+      .set('x-elastic-internal-origin', 'foo')
+      .send({
+        name: 'test Open AI',
+        connector_type_id: '.gen-ai',
+        config,
+        secrets: {
+          apiKey: 'genAiApiKey',
+        },
+      })
+      .expect(200)
+  ).body;
+
+  yield connector;
+}
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common', 'searchPlayground']);
+  const configService = getService('config');
+  const supertest = getService('supertest');
```

Comment:
There is a problem if you wish to run the test for both ESS and Serverless, because for ESS you can make API calls with basic auth as you defined here, but for serverless you have to use api keys generated with SAML auth.
Please check the docs https://github.com/elastic/kibana/tree/main/x-pack/test_serverless#api-integration-test-example

Reply:
Should we follow the guide for API integration tests if there are functional tests? The API I use is to define an environment that the tested package is using.

I mean, I use supertest to define a connector. Maybe there is something more appropriate to use for it?